torch/_inductor/codecache.py

```diff
@@ -147,79 +147,176 @@ def is_gcc():
+                    sysconfig.get_config_var("LIBDIR")
+                ]
+                env["LD_LIBRARY_PATH"] = ":".join(lpaths)
+                # Check dry run result
+                subprocess.check_call(build_cmd, stderr=subprocess.STDOUT, env=env)
+            except Exception as e:
+                return False
+
+            return True
+
+
+@dataclasses.dataclass
+class VecAVX512(VecISA):
+    _bit_width = 512
+    _macro = "CPU_CAPABILITY_AVX512"
+    _arch_flags = "-mavx512f -mavx512dq -mavx512vl -mavx512bw -mfma"
+    _dtype_nelements = {torch.float: 16, torch.bfloat16: 32}
+
+    def __str__(self) -> str:
+        return "avx512"
+
+    __hash__: Callable[[VecISA], Any] = VecISA.__hash__
+
+
+@dataclasses.dataclass
+class VecAVX2(VecISA):
+    _bit_width = 256
+    _macro = "CPU_CAPABILITY_AVX2"
+    _arch_flags = "-mavx2 -mfma"
+    _dtype_nelements = {torch.float: 8, torch.bfloat16: 16}
+
+    def __str__(self) -> str:
+        return "avx2"
+
+    __hash__: Callable[[VecISA], Any] = VecISA.__hash__
+
+
+class InvalidVecISA(VecISA):
+    _bit_width = 0
+    _macro = ""
+    _arch_flags = ""
+    _dtype_nelements = {}
+
+    def __str__(self) -> str:
+        return "INVALID_VEC_ISA"
 
     def __bool__(self):
-        return self != _SupportedVecIsa.INVALID
+        return False
 
-    @staticmethod
-    def isa_str(supported_isa: enum.Enum):
-        if supported_isa == _SupportedVecIsa.AVX512:
-            return "avx512"
-        elif supported_isa == _SupportedVecIsa.AVX2:
-            return "avx2"
-        else:
-            return ""
+    __hash__: Callable[[VecISA], Any] = VecISA.__hash__
 
-    @staticmethod
-    def vec_macro(supported_isa: enum.Enum):
-        if supported_isa == _SupportedVecIsa.AVX512:
-            return "CPU_CAPABILITY_AVX512"
-        elif supported_isa == _SupportedVecIsa.AVX2:
-            return "CPU_CAPABILITY_AVX2"
-        else:
-            return ""
+
+invalid_vec_isa = InvalidVecISA()
+supported_vec_isa = [VecAVX512(), VecAVX2()]
 
 
 # Cache the cpuinfo to avoid I/O overhead. Meanwhile, the cpuinfo content
 # might have too much redundant content that is useless for ISA check. Hence,
 # we only cache some key isa information.
-@functools.lru_cache(1)
-def get_cpu_proc_info():
+@functools.lru_cache(None)
+def valid_vec_isa():
     if sys.platform != "linux":
         return []
 
-    isa_info = []
+    isa_list = []
     with open("/proc/cpuinfo") as _cpu_info:
         _cpu_info_content = _cpu_info.read()
-        if _SupportedVecIsa.isa_str(_SupportedVecIsa.AVX512) in _cpu_info_content:
-            isa_info.append(_SupportedVecIsa.AVX512)
+        for isa in supported_vec_isa:
+            if str(isa) in _cpu_info_content and isa:
+                isa_list.append(isa)
+        return isa_list
 
-        if _SupportedVecIsa.isa_str(_SupportedVecIsa.AVX2) in _cpu_info_content:
-            isa_info.append(_SupportedVecIsa.AVX2)
 
-        return isa_info
+def pick_vec_isa():
```

Comment:
`preferred_vec_isa` or maybe `current_vec_isa`?

Reply:
The style of the function name is similar with `pick_loop_order` https://github.com/pytorch/pytorch/blob/master/torch/_inductor/ir.py#LL1971C22-L1971C22